launcher/src/main/java/org/apache/brooklyn/launcher/BrooklynWebServer.java

```diff
@@ -447,7 +450,15 @@ public synchronized void start() throws Exception {
         rootContext.setTempDirectory(Os.mkdirs(new File(webappTempDir, "war-root")));
 
         server.setHandler(handlers);
-        server.start();
+        try {
+            server.start();
+        } catch (BindException e) {
+            // port discovery routines may take some time to clear, e.g. 250ms for SO_TIMEOUT
+            // tests fail because of this; see if adding a delay improves things
+            log.warn("Initial server start-up failed binding (retrying after a delay): "+e);
+            Time.sleep(Duration.millis(500));
+            server.start();
```

Comment:
@neykov Why would `setReuseAddress(true)` improve false positives?  As long as we make the same call when we start the server it should be identical?  (Or are you thinking that the root cause is that the server-start call isn't using that, and that is what you'd fix?)

I agree finding and fixing the root cause (and recognising and logging better when we encounter it?) is even better, but I don't know what it is!

I don't think we do parallel tests so I suspect that isn't the issue.  Letting sockets pick their own ports is a big change so I'd rather not go down that line.